File: Stronghold/Stronghold.Application/DTOs/Response/CheckoutResponse.cs
namespace Stronghold.Application.DTOs.Response
{
    public class CheckoutResponse
    {
        public string ClientSecret { get; set; } = string.Empty;
        public string PaymentIntentId { get; set; } = string.Empty;
        public decimal TotalAmount { get; set; }
    }
}

File: packages/stronghold_core/lib/models/responses/checkout_response.dart
/// Checkout response with Stripe payment intent details
class CheckoutResponse {
  final String clientSecret;
  final String paymentIntentId;
  final double totalAmount;

  const CheckoutResponse({
    required this.clientSecret,
    required this.paymentIntentId,
    required this.totalAmount,
  });

  factory CheckoutResponse.fromJson(Map<String, dynamic> json) {
    return CheckoutResponse(
      clientSecret: json['clientSecret'] as String,
      paymentIntentId: json['paymentIntentId'] as String,
      totalAmount: (json['totalAmount'] as num).toDouble(),
    );
  }
}
